speechbrain/core.py

```diff
@@ -1715,22 +1715,20 @@ def _wrap_distributed(self):
         if not self.distributed_launch and not self.data_parallel_backend:
             return
         elif self.distributed_launch:
-            for name, module in self.modules.items():
-                if any(p.requires_grad for p in module.parameters()):
-                    module = SyncBatchNorm.convert_sync_batchnorm(module)
-                    if self.distributed_backend == "gloo":
-                        module = DDP(
-                            module,
-                            device_ids=None,
-                            find_unused_parameters=self.find_unused_parameters,
-                        )
-                    else:
-                        module = DDP(
-                            module,
-                            device_ids=[self.device],
-                            find_unused_parameters=self.find_unused_parameters,
-                        )
-                    self.modules[name] = module
+            self.modules = SyncBatchNorm.convert_sync_batchnorm(self.modules)
+            if self.distributed_backend == "gloo":
+                self.modules = DDP(
+                    self.modules,
+                    device_ids=None,
+                    find_unused_parameters=self.find_unused_parameters,
+                )
+            else:
+                self.modules = DDP(
+                    self.modules,
+                    device_ids=[self.device],
+                    find_unused_parameters=self.find_unused_parameters,
+                )
+            self.modules = self.modules.module
```

Comment:
What is the purpose of this change? from `self.modules.module` to `self.modules`

Is it related to the fix for DDP? Because I thought removing the line above:

```
                if any(p.requires_grad for p in module.parameters()):
```

was the main fix here.

Reply:
If you don't do this, then you can't do self.modules.Transformer. But you will need to do self.modules.module.Transformer... DDP wrapping ass an extra .module I.e refactor 100% of the recipes

Reply:
We have changed of strategy (see above and commits)